### ISSUE TITLE
some files appear broken at download and won't install

### ISSUE THREAD

**Issue by holtzermann17** [melpa/melpa#7070]
For example:
```
Install package ‘julia-mode-20200717.1915’? (y or n) y
Contacting host: melpa.milkbox.net:80
Parsing tar file... 
Warning: premature EOF parsing tar file
```

This looks like a similar issue:
https://github.com/xuchunyang/elisp-demos/issues/3

The following packages are affected:
(cider-20200718.1908, company-20200616.2354, dash-20200524.1947, ejc-sql-20200716.854, git-commit-20200701.2112, gruvbox-theme-20200622.58, helm-20200722.1531, helm-core-20200722.1046, julia-mode-20200717.1915, julia-repl-20200625.924, magit-20200723.236, org-roam-20200722.1259, org-roam-bibtex-20200716.1450, transient-20200719.955, use-package-20200721.2156, with-editor-20200720.2014)

My system: 

GNU Emacs 28.0.50 (build 1, x86_64-pc-linux-gnu, GTK+ Version 3.22.30, cairo version 1.15.10) of 2020-08-11

**Comment by holtzermann17** [comment]
Update: I see this blob of HTML in the individual package buffers:

```html
<!DOCTYPE html PUBLIC "-//W3C//DTD XHTML 1.0 Transitional//EN" "http://www.w3.org/TR/xhtml1/DTD/xhtml1-transitional.dtd"><html><head><meta http-equiv="refresh" content="0;url=http://error.talktalk.co.uk/main?ParticipantID=nmum6rqpq5q6gkm3933n3nf9s76onu6r&FailedURI=http%3A%2F%2Fmelpa.milkbox.net%2Fpackages%2Fdash-readme.txt&FailureMode=1&Implementation=&AddInType=4&Version=pywr1.0&ClientLocation=uk"/><script type="text/javascript">url="http://error.talktalk.co.uk/main?ParticipantID=nmum6rqpq5q6gkm3933n3nf9s76onu6r&FailedURI=http%3A%2F%2Fmelpa.milkbox.net%2Fpackages%2Fdash-readme.txt&FailureMode=1&Implementation=&AddInType=4&Version=pywr1.0&ClientLocation=uk";if(top.location!=location){var w=window,d=document,e=d.documentElement,b=d.body,x=w.innerWidth||e.clientWidth||b.clientWidth,y=w.innerHeight||e.clientHeight||b.clientHeight;url+="&w="+x+"&h="+y;}window.location.replace(url);</script></head><body></body></html>
```

This suggests that there's something going on with my ISP (Talk Talk).

In full:

```
Package dash is available.

     Status: Available from melpa -- Install
    Archive: melpa
    Version: 20200524.1947
     Commit: 732d92eac56023a4fb4a5dc3d9d4e274ebf44bf9
    Summary: A modern list library for Emacs
Required by: org-roam-20200422.635, org-alert-20180524.133, magit-20200208.551, git-commit-20200207.1819, f-20191110.1357, ejc-sql-20200212.1038,
               dash-functional-20200617.702, company-org-roam-20200711.355, bibtex-completion-20200513.852, biblio-core-20200416.307,
               autothemer-20180920.923
   Keywords: lists 
 Maintainer: Magnar Sveen <magnars@gmail.com>
     Author: Magnar Sveen <magnars@gmail.com>
Other versions: 20200119.2310 (installed), 2.12.0 (gnu).

<!DOCTYPE html PUBLIC "-//W3C//DTD XHTML 1.0 Transitional//EN" "http://www.w3.org/TR/xhtml1/DTD/xhtml1-transitional.dtd"><html><head><meta http-equiv="refresh" content="0;url=http://error.talktalk.co.uk/main?ParticipantID=nmum6rqpq5q6gkm3933n3nf9s76onu6r&FailedURI=http%3A%2F%2Fmelpa.milkbox.net%2Fpackages%2Fdash-readme.txt&FailureMode=1&Implementation=&AddInType=4&Version=pywr1.0&ClientLocation=uk"/><script type="text/javascript">url="http://error.talktalk.co.uk/main?ParticipantID=nmum6rqpq5q6gkm3933n3nf9s76onu6r&FailedURI=http%3A%2F%2Fmelpa.milkbox.net%2Fpackages%2Fdash-readme.txt&FailureMode=1&Implementation=&AddInType=4&Version=pywr1.0&ClientLocation=uk";if(top.location!=location){var w=window,d=document,e=d.documentElement,b=d.body,x=w.innerWidth||e.clientWidth||b.clientWidth,y=w.innerHeight||e.clientHeight||b.clientHeight;url+="&w="+x+"&h="+y;}window.location.replace(url);</script></head><body></body></html>

[back]
```

**Comment by purcell** [comment]
Couple of things I notice here: you're connecting to a very old MELPA domain name (melpa.milkbox.net) and you're connecting on port 80 (http). You should be using https and melpa.org these days, so try updating your `package-archives` config as per the current README.

**Comment by purcell** [comment]
I've closed this because I'm 99.9% confident this will resolve your issue, but correct me if not.
